include/oneapi/dpl/internal/binary_search_impl.h

```diff
@@ -188,10 +226,25 @@ binary_search_impl(Policy&& policy, InputIterator1 start, InputIterator1 end, In
     auto keep_result = oneapi::dpl::__ranges::__get_sycl_range<__bknd::access_mode::read_write, OutputIterator>();
     auto result_buf = keep_result(result, result + value_size);
     auto zip_vw = make_zip_view(input_buf.all_view(), value_buf.all_view(), result_buf.all_view());
-    __bknd::__parallel_for(::std::forward<Policy>(policy),
-                           custom_brick<StrictWeakOrdering, decltype(size), binary_search>{comp, size}, value_size,
-                           zip_vw)
-        .wait();
+
+    // Enable index calculation to proceed with uint32_t if input range is small enough.
+    if (size <= ::std::numeric_limits<::std::uint32_t>::max())
+    {
+        __bknd::__parallel_for(::std::forward<Policy>(policy),
+                               custom_brick<StrictWeakOrdering, ::std::uint32_t, search_algorithm::binary_search>{
+                                   comp, static_cast<::std::uint32_t>(size)},
+                               value_size, zip_vw)
+            .wait();
+    }
+    else
+    {
+        auto fallback_policy = __bknd::make_wrapped_policy<binary_search_fallback>(::std::forward<Policy>(policy));
```

Comment:
Do we really need to have `fallback_policy` as local variable?

Reply:
No, I have changed this.